libcaf_core/caf/typed_actor.hpp

```diff
@@ -249,11 +260,59 @@ class typed_actor : detail::comparable<typed_actor<Sigs...>>,
     return ptr_.release();
   }
 
-  typed_actor(actor_control_block* ptr) : ptr_(ptr) {
+  strong_actor_ptr ptr_;
+};
+
+/// Identifies a statically typed actor.
+/// @tparam Ts Signatures of all accepted messages.
+/// @note This is a specialization for backwards compatibility with pre v1.0
+/// releases. Please use the trait based implementation.
+template <class T1, class T2, class... Ts>
+class typed_actor<T1, T2, Ts...>
+  : public typed_actor<statically_typed<T1, T2, Ts...>> {
+public:
+  template <class...>
+  friend class typed_actor;
+
+  using super = typed_actor<statically_typed<T1, T2, Ts...>>;
+
+  using behavior_type = typed_behavior<T1, T2, Ts...>;
+
+  /// The default, event-based type for implementing this messaging interface.
+  using impl = typed_event_based_actor<T1, T2, Ts...>;
+
+  /// Identifies pointers to instances of this kind of actor.
+  using pointer = impl*;
+
+  /// A class type suitable as base type class-based implementations.
+  using base = impl;
+
+  /// The default, event-based type for implementing this messaging interface as
+  /// a stateful actor.
+  template <class State>
+  using stateful_impl = stateful_actor<State, impl>;
+
+  /// Convenience alias for `stateful_impl<State>*`.
+  template <class State>
+  using stateful_pointer = stateful_impl<State>*;
+
+  /// A view to an actor that implements this messaging interface without
+  /// knowledge of the actual type.
+  using pointer_view = typed_actor_pointer<T1, T2, Ts...>;
+
+  using super::super;
+
+  typed_actor(const super& other) : super(other) {
+    // nop
+  }
+  typed_actor(super&& other) : super(std::move(other)) {
     // nop
   }
 
-  strong_actor_ptr ptr_;
+  typed_actor& operator=(std::nullptr_t) {
```

Comment:
Shouldn't we also default `const typed_actor&` and `typed_actor&&` overloads?

Reply:
I've added a static assert to test this. Not sure how it works, I think the implementation falls back to the base class, by binding a `const&` to it's `operator=`.